### ISSUE TITLE
Extract fetchYaml and maybe other fetch functions

### ISSUE THREAD

**Issue by paulmelnikow** [badges/shields#2518]
See discussions:

- https://github.com/badges/shields/pull/2487#discussion_r241169902
- https://github.com/badges/shields/pull/2399#issuecomment-444970457

**Comment by paulmelnikow** [comment]
Should we keep this open?

The thing we could still do is delegate from `_requestJson`, `_requestYaml`, etc. to functions which can be invoked on any instance of `BaseService`.

**Comment by chris48s** [comment]
Aah yes. I was unsure on this. My initial thought was we could extract them to stand-alone functions. The reason I didn't do this was because inside the `_requestFormat()` functions we're calling `this._request()` and `this.constructor._validate()` so `_requestFormat()` doesn't extract so nicely outside of the class.

We could do it so that we pass in a `requester` function and a `validator` function as params, or so that we can pass an instance of `BaseService` as a param. Another option might be to try and do it as a [mixin](https://javascript.info/mixins). I'm not really sure which of those makes the most/least sense.

Given the F-Droid badge could just extend `BaseYamlService` I decided to just leave it and deal with it once we've got an example where we actually need this abstraction.
